### ISSUE TITLE
Injection not applied to command parameter

### ISSUE THREAD

**Issue by t-gehring** [jeff-hykin/macro-commander#2]
The injection parameter is applied to the "args" and "hiddenConsole" parameters of a macro step, but isn't currently applied to the command parameter. Adding this would be a small code change, but allows for much more flexibility in macro creation.

As an example, the following macro:
```json
"terminal.args": [
    {
        "injections": [
            {
                "replace": "$args",
                "withResultOf": "var succeeded = false; window.showInputBox().then((input) => {if (input != undefined) {succeeded = true}; return input;})"
            },
        ],
        "command": "workbench.action.terminal.sendSequence",
        "args": {"text": " $args"}
    },
    "workbench.action.terminal.focus",
    {
        "injections": [{"replace": "$nextAction", "withResultOf": "if (succeeded) \"sendSequence\"; else \"kill\""}],
        "command": "workbench.action.terminal.$nextAction",
        "args": {"text": "\n"}
    },
]
```
raises an input box for the user to enter parameters for a command that has been placed in the terminal, but then closes the terminal pane if the user presses ESC to cancel input.

**Comment by jeff-hykin** [comment]
Yeah I think I can make this change.

If I remember how I coded it correctly, the `var succeeded = false` in the first statement won't be accessible later in the `if (succeeded)` part. The reason being that, the first part is wrapped inside of a function (otherwise the `return input` of the first part wouldn't work)

What I can do though is provide an `external` variable so you can do `external.succeeded = true` and then later `if (external.succeeded)`

**Comment by t-gehring** [comment]
The JavaScript commands are wrapped in a function, but the injections are just eval'd, so currently they can leak variables. I'm certain of that, because I patched the extension locally to do replacements in the command, and had that exact macro running correctly.

But providing an external variable for both JavaScript commands and injections would be really handy, and much neater than abusing a leaky eval in this case. It would also let you do more complex macros which mix js and vs commands.

I'd recommend making that external variable shared for all macros, rather than local to a particular macro execution - that lets you do cool things like having a macro re-raise a terminal it created last time, instead of constantly spawning new ones (for example).

**Comment by t-gehring** [comment]
Or if you want to provide better encapsulation, you could make each `external` object local to the macro execution, but give it a `shared` member which is from a shared scope. That way a macro could have it's own `external.success` flags which don't trample over another macro's data, but you can still pass data between different macros, and different calls of the same macro.

**Comment by jeff-hykin** [comment]
Thats interesting, I didn't know eval would allow for a top-level return statement like that. I think those are good ideas.

> like having a macro re-raise a terminal it created last time, instead of constantly spawning new ones (for example)

That would be nice, although it is hard to test to make sure the user didn't delete it.

I'll probably make one variable, like `macros` and then put all of the special functionality inside of it. It can work like a namespace, and also make Object.keys(macros) a kind of way to see all of the available features.

It might take me a little bit to get around to making these changes, I'm currently trying to wrap up several other projects. I also want to clean up the extension in general and make it a bit more like [Command Runner](https://marketplace.visualstudio.com/items?itemName=edonet.vscode-command-runner).

**Comment by t-gehring** [comment]
>  I didn't know eval would allow for a top-level return statement like that

In my example macro the return is actually from the function running as the `then` of the `showInputBox()`'s promise, but the `succeeded` variable is defined in the main scope, which is placed in the same scope as the rest of the macro execution it seems. I think a top-level return in the `eval` would cause issues like you expect.

I'll try to put together a PR for just replacing in the command, and to fix Issue #4, and I'll let you figure out the rest how you want it. Those two additions, plus the leaky `eval` hack, are plenty enough to do what I'm trying to do atm.

And while I'm commenting - thank you so much for making this extension! It seems crazy that VS Code has no simple way to make small scripts like this without building a full extension yourself, and macro commander hits a nice medium of simple enough to get something going quickly, and complex enough to do very interesting things.

**Comment by jeff-hykin** [comment]
> the return is actually from the function running as the `then`

Ah, I was reading on my phone and didn't scroll far enough to see the closing `;})`. 

> I'll try to put together a PR for just replacing in the command, and to fix Issue #4, and I'll let you figure out the rest how you want it. 

Thanks, I appreciate it. #4 will be a huge improvement.

> It seems crazy that VS Code has no simple way

Yeah its surprising, there's lots of things like this. I had to fork [an extension to display line endings](https://marketplace.visualstudio.com/items?itemName=jeff-hykin.code-eol), and get [an extension](https://marketplace.visualstudio.com/items?itemName=Tyriar.vscode-terminal-here) to just open a terminal in the same directory as the currently-open file. Code Runner and Project Manager are much more popular extensions that also seem like they are standard/basic features.

I've even thought about making a collection called "VS Code Userland" of all the extensions that are simple generic quality-of-life features.

**Comment by jeff-hykin** [comment]
Took a little while lol, but I finally created the `sharedMacroInfo` variable for between-macro communication. The other thing this issue was about was solved earlier too!
